src/modules/student/dto/update-student.dto.ts

```diff
@@ -0,0 +1,32 @@
+import { PartialType } from '@nestjs/mapped-types';
+import { CreateStudentDto } from './create-student.dto';
+import { ContractType, TypeWork } from 'src/types';
+
+export class UpdateStudentDto extends PartialType(CreateStudentDto) {
```

Comment:
tutaj jest to niepotrzebne bo UpdateStudentDto bierze cały CreateStudentDto i daje wszystkie pola jako opcjonalne

Reply:
czyli rozumiem, że UpdateStudentDto można wywalić a CreateStudentDto użyć też do update? 
Jeśli tak to oczywiście zmienię nazwę CreateStudentDto na StudentDto

Reply:
nie, update zostaje ale wystarczy że będzie tak:
export class UpdateStudentDto extends PartialType(CreateStudentDto) {}